lesson1/exercise4.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Exercise 4: What is all this stuff"
+nav_order: 5
+parent: 1 - First Steps with Java
+---
+
+# Exercise 4: What is all this stuff
+
+We typed or copy-pasted bits of Java code without fully understanding what certain words or certain brackets mean. Here we drop the curtain and show the meaning behind that text.
+
+## Objective
+
+In this exercise we will learn about:
+ * Different kinds of brackets
+ * Keywords
+ * Importance of structure
+
+At the end of this you should understand:
+ * How do we structure code using the curly brackets
+ * How and where are round brackets used
+ * What are keywords and provide examples of some Java keywords
+ * What is and why do we structure programs
+
+## Brackets
+
+Programming has a deep connection with Mathematics, first computing devices were built to solve simple math problems such as counting items or simple arithmetic. That's why there's a great deal of brackets that are in use -- they come from math.
+
+**INFO:** If you're interested in more on this, look up the article on the [History of computing hardware](https://en.wikipedia.org/wiki/History_of_computing_hardware) on Wikipedia, and this [question and answer](https://softwareengineering.stackexchange.com/questions/136987/what-does-mathematics-have-to-do-with-programming) on StackExchange.
+
+Here are the types of brackets used in Java (and many similar programming languages).
+
+| Bracket             | Name            | Use                                                                                             |
+|:--------------------|:----------------|:------------------------------------------------------------------------------------------------|
+| **`{`** and **`}`** | Curly brackets  | Organize code into blocks, we'll learn about different blocks of code in later lessons          |
+| **`(`** and **`)`** | Round brackets  | Used to group parameters and expressions, think _f(x) = x + 1_                                  |
+| **`[`** and **`]`** | Square brackets | Used to declare and access arrays, arrays are way to work with multiple values of the same type |
+
+### Curly brackets organize code
+
+We often need to organize code, or in other words, we give the source code structure. We do this with blocks of code that start with an opening curly bracket: **`{`** and ending with a closing curly bracket **`}`**.
```

Comment:
At this point, let's spend a couple of words about indentation.

Reply:
actually I looked at wikipedia (https://en.wikipedia.org/wiki/Bracket#Names_for_various_bracket_symbols) and there are apparently other terms for brackets (depending on if you use e.g. British or American English).
especially terms I used and heard in past like parentheses for round brackets or e.g. braces instead of brackets.

I wonder: should we add these alternative names also here or would that be too confusing? 
On the other hand, we as teachers then must take extra care to not use an of the alternative words ...

Reply:
I've added the link to Wikipedia, I think we should be consistent but I'm sure we'll slip without even noticing.